src/IK/src/QPInverseKinematics.cpp

```diff
@@ -0,0 +1,405 @@
+                         "the following description {}.",
+                         logPrefix,
+                         name,
+                         task.task->getDescription());
+            return false;
+        }
+
+        if (task.priority == 0)
+        {
+            numberOfConstraints += task.task->size();
+        }
+    }
+
+    m_pimpl->solver.data()->setNumberOfVariables(handler.getNumberOfVariables());
+    m_pimpl->solver.data()->setNumberOfConstraints(numberOfConstraints);
+
+    // resize matrices
+    m_pimpl->hessian.resize(handler.getNumberOfVariables(), handler.getNumberOfVariables());
+    m_pimpl->gradient.resize(handler.getNumberOfVariables());
+
+    m_pimpl->constraintMatrix.resize(numberOfConstraints, handler.getNumberOfVariables());
+    m_pimpl->upperBound.resize(numberOfConstraints);
+    m_pimpl->lowerBound.resize(numberOfConstraints);
+
+    if (!handler.getVariable(m_pimpl->robotVelocityVariable.name,
+                             m_pimpl->robotVelocityVariable))
+    {
+        log()->error("{} Error while retrieving the robot velocity variable.", logPrefix);
+        return false;
+    }
+
+    return true;
+}
+
+std::vector<std::string> QPInverseKinematics::getTasksName() const
+{
+    std::vector<std::string> tasksName;
+
+    for (const auto& [name, task] : m_pimpl->tasks)
+    {
+        tasksName.push_back(name);
+    }
+
+    return tasksName;
+}
+
+bool QPInverseKinematics::advance()
+{
+    constexpr auto logPrefix = "[QPInverseKinematics::advance]";
+
+    // when advance is called the previous solution is no more valid
+    m_pimpl->isValid = false;
+
+    if (!m_pimpl->isInitialized)
+    {
+        log()->error("{} Please call initialize() before advance().", logPrefix);
+        return false;
+    }
+
+    // update of all the tasks
+    for (auto& [name, task] : m_pimpl->tasks)
+    {
+        task.task->update();
+    }
+
+    // Compute the gradient and the hessian
+    m_pimpl->hessian.setZero();
+    m_pimpl->gradient.setZero();
+    for (const auto& cost : m_pimpl->costs)
+    {
+        Eigen::Ref<const Eigen::MatrixXd> A = cost.get().task->getA();
+        Eigen::Ref<const Eigen::VectorXd> b = cost.get().task->getB();
+
+        m_pimpl->hessian += A.transpose() * cost.get().weight.asDiagonal() * A;
+        m_pimpl->gradient -= A.transpose() * cost.get().weight.asDiagonal() * b;
+    }
+
+    // compute the constraints
+    std::size_t index = 0;
+    for (const auto& constraint : m_pimpl->equalityConstraints)
+    {
+        Eigen::Ref<const Eigen::MatrixXd> A = constraint.get().task->getA();
+        Eigen::Ref<const Eigen::VectorXd> b = constraint.get().task->getB();
+
+        m_pimpl->lowerBound.segment(index, constraint.get().task->size()) = b;
+        m_pimpl->upperBound.segment(index, constraint.get().task->size()) = b;
+        m_pimpl->constraintMatrix.middleRows(index, constraint.get().task->size()) = A;
+
+        index += constraint.get().task->size();
```

Comment:
Note that it would be enough to check if at the end ``index`` is equal to the total number of constraints. You cannot know which constraint increased in dimension, but you know that at least one did.